### PR TITLE
[codex] Harden omitted-ierr lifecycle errors

### LINKED ISSUE #126: Revisit permissive force-stop and recovery behavior when ierr is omitted
## Summary
The library should revisit whether omitted-`ierr` recovery paths are too permissive when active timers remain or mismatches occur.

## Why this matters
A correctness-first timing library may be better served by failing louder in some cases rather than force-stopping state and producing believable-but-synthetic results after a user mistake.

## Evidence from the assessment
- `init`, `reset`, and `finalize` can force-stop active timers when `ierr` is omitted.
- Warn-mode mismatch handling repairs after emitting an error.
- In larger MPI jobs, stderr-heavy permissive recovery can be operationally messy.

## Related issues
- Umbrella: #129

## Possible directions
- Revisit which cases should stay permissive versus fail harder.
- Add an explicit strict-no-recovery mode for operational use.
- Document the tradeoff more aggressively if the current behavior is intentional.

### PR DESCRIPTION
**Pull request by jaharris87**
## Summary
- stop `init`, `reset`, and `finalize` from force-stopping active timers when `ierr` is omitted
- route active lifecycle failures through the standard status reporter so `ierr` callers still get `FTIMER_ERR_ACTIVE` while omitted-`ierr` callers get a warning and no state mutation
- update edge-case tests and current-state docs to pin down the non-mutating lifecycle contract and the explicit mismatch-repair exception

## Why
Omitted-`ierr` lifecycle misuse previously force-stopped active timers and performed hidden cleanup, which could synthesize believable timing data after a caller error. This change makes those paths correctness-first again: they fail loudly without mutating runtime state.

## User Impact
- omitted-`ierr` calls to `init`, `reset`, or `finalize` while timers are active now warn and return with the timer set unchanged
- callers that pass `ierr` still receive `FTIMER_ERR_ACTIVE` with no stderr output
- stop-mismatch repair remains available only through explicit mismatch modes

## Validation
- `cmake --build build-tests`
- `ctest --test-dir build-tests --output-on-failure -R ftimer_serial_tests`
- `ctest --test-dir build-tests --output-on-failure`

Closes #126
Refs #129

### CODE REVIEW

**Comment by jaharris87** [comment]
@codex review as a senior software engineer reviewing this fTimer PR for correctness; inspect changed code, changed tests, and changed user-facing docs, then find behavior regressions, wrapper or OOP parity drift, install or export breakage, unsafe defaults, silent lossy behavior, summary or output regressions, and missing end-to-end validation, not style issues; when timer runtime paths change, explicitly sanity-check start or stop ordering, context attribution, and docs/tests parity; classify findings as bug, test gap, design concern, or nit; begin with "## Software Review".

**Comment by jaharris87** [comment]
@codex review as an fTimer timing and HPC methodology reviewer; verify that stop pops the call stack before context lookup, mismatch repair uses one timestamp, repair suppresses callbacks, self-time subtracts only direct children, timing arithmetic stays in real(wp), the error contract stays honest, and MPI or OpenMP semantics remain correct; report only real flaws, not style issues; classify findings as timing error, stack corruption, precision loss, MPI safety, error contract violation, or methodology concern; begin with "## Methodology Review".

File: tests/test_edge_cases.pf [hunk 1]
```diff
@@ -130,23 +131,134 @@ contains
...
+      @assertTrue(state%segments(1)%is_running(ctx))
+      @assertEqual(0.0_wp, state%segments(1)%time(ctx))
+      @assertEqual(1, state%segments(1)%call_count(ctx))
+
+      fake_time = 3.0_wp
+      call timer%stop("A", ierr=ierr)
+      @assertEqual(FTIMER_SUCCESS, ierr)
+      call timer%get_summary(summary, ierr=ierr)
+      @assertEqual(FTIMER_SUCCESS, ierr)
+      @assertEqual(3.0_wp, summary%total_time)
```
[the remark this answers is not in the document]
Comment (jaharris87) [reply tests/test_edge_cases.pf]: Addressed in `a6294b4`. I extended the finalize warning-path test to run a nested `outer -> B` and `outer -> C` sequence after the warning-return and assert per-entry inclusive/self times, so the test now exercises nested stop ordering and sibling-boundary self-time behavior instead of only a single root timer.
